packages/devtools_app/lib/src/primitives/auto_dispose.dart

```diff
@@ -75,6 +76,35 @@ class Disposer {
     }
     _focusNodes.clear();
   }
+
+  /// Runs [callback] when [trigger]'s value satisfies the [readyWhen] function.
+  ///
+  /// When calling [callOnceWhenReady] :
+  ///     - If [trigger]'s value satisfies [readyWhen], then the [callback] will
+  ///       be immediately triggered.
+  ///     - Otherwise, the [callback] will be triggered when [trigger]'s value
+  ///       changes to equal [readyWhen].
+  ///
+  /// Any listeners set by [callOnceWhenReady] will auto dispose, or be removed after the callback is run.
+  void callOnceWhenReady<T>({
+    required VoidCallback callback,
+    required ValueListenable<T> trigger,
+    required bool Function(T triggerValue) readyWhen,
+  }) {
+    if (readyWhen(trigger.value)) {
+      callback();
+    } else {
+      // do the stuff to add the listener and remove it when appropriate
+      VoidCallback? earlyDisposeCallback;
```

Comment:
nit: rename to `listener` or `triggerListener`. I think the "early dispose" part is a bit confusing.